packages/rancher-monitoring/generated-changes/patch/values.yaml.patch

```diff
@@ -230,41 +245,32 @@
 +      - effect: "NoSchedule"
 +        operator: "Exists"
 +
-+## Component scraping nginx-ingress-controller
-+##
-+ingressNginx:
++rke2IngressNginx:
 +  enabled: false
-+
-+  ## The namespace to search for your nginx-ingress-controller
-+  ##
-+  namespace: ingress-nginx
-+  
-+  service:
-+    port: 9913
-+    targetPort: 10254
-+    # selector:
-+    #   app: ingress-nginx
-+  serviceMonitor:
-+    ## Scrape interval. If not set, the Prometheus default scrape interval is used.
-+    ##
-+    interval: ""
-+
-+    ## 	metric relabel configs to apply to samples before ingestion.
-+    ##
-+    metricRelabelings: []
-+    # - action: keep
-+    #   regex: 'kube_(daemonset|deployment|pod|namespace|node|statefulset).+'
-+    #   sourceLabels: [__name__]
-+
-+    # 	relabel configs to apply to samples before ingestion.
-+    ##
-+    relabelings: []
-+    # - sourceLabels: [__meta_kubernetes_pod_node_name]
-+    #   separator: ;
-+    #   regex: ^(.*)$
-+    #   targetLabel: nodename
-+    #   replacement: $1
-+    #   action: replace
++  metricsPort: 10254
++  component: ingress-nginx
++  clients:
++    port: 10015
++    useLocalhost: true
++    deployment:
++      replicas: 1
++      affinity:
++        podAffinity:
++          requiredDuringSchedulingIgnoredDuringExecution:
++            - labelSelector:
++                matchExpressions:
++                  - key: "app.kubernetes.io/component"
++                    operator: "In"
++                    values:
++                      - "controller"
++              topologyKey: "kubernetes.io/hostname"
++              namespaces:
++                - "kube-system"
```

Comment:
The nginx-controller is 
- a Deployment with 1 pod in RKE2 cluster which version <= 1.20
- a Daemonset in RKE2 cluster which version since 1.21

The podAffinity makes the pod to be assigned to the node where the nginx-controller's pod runs.

Reply:
Can you add this as a comment to the values.yaml itself?

Also, do we need to specify this in the UI ticket to infer what the value of `rke2IngressNginx.clients.deployment.[replicas|enabled]?` should be based on Cluster Type + k8s version? I'm not sure we support that today. 

What happens if a user has RKE 1.20 cluster and upgrades to a RKE 1.21 cluster? Do we need to add documentation that Monitoring needs to be updated in these situations? 

It slightly worries me since transitioning from a Deployment -> DaemonSet in rke2 k8s 1.21 wouldn't actually show any targets down on the UI on an upgrade, even though you are completely omitting metrics in the latter case.

Maybe the right option here would be to set `rke2IngressNginx.clients.deployment.[replicas|enabled]=[0|false]` by default, which would still work since the affinityRules should work on a DaemonSet (but will leave pods pending), and specify in the docs that an RKE2 cluster with a version <=1.20 can choose to override the deployment flag to prevent Pods from being stuck on Pending (with a giant warning sign that says that they must upgrade monitoring on a k8s upgrade).